### PR TITLE
Valid move pawn

### PR DESCRIPTION
**Pull request by dmk1991**
# Implements valid_move? for the pawn piece.
The method checks three things,
- If the piece is on it's starting spot, it can move either one or two steps.
- If the piece is taking a piece, it can move one diagonal.
- If the piece has already moved, it can only ever move one step ever again.

Let me know what you guys think, looking for feedback on the general structure of the if / elsif / elsif.

### CODE REVIEW

**Review by iandelible — COMMENTED** [review]
1. Do we need to add is_obstructed? inside the valid_move?
2. Line 11 is accounting for capturing?
3. Should we write a test to test that that a pawn doesn't move backwards?

File: app/models/pawn.rb [hunk 1]
```diff
@@ -1,2 +1,19 @@
...
+  def valid_move?(row_destination, col_destination)
+    start_row = self.is_black? ? 7 : 2
+    direction = self.is_black ? 1 : -1
+
+    row_distance = self.row - row_destination
+    col_distance = self.column - col_destination
+
+    if self.row == start_row && self.column == col_destination
+      return row_distance == 2 * direction || row_distance == 1 * direction
+    elsif row_distance == 1 * direction && col_distance == 1 * direction
```
Comment (jacobwgillespie) [inline app/models/pawn.rb]: I think `col_distance == 1 * direction` will limit capture direction (left or right) based on the color of the piece.  You likely want `col_distance.abs == 1`

👍

Comment (dmk1991) [reply app/models/pawn.rb]: I think you are correct Jacob, I'll get a fix and tests up asap gentlemen.

**Comment by dmk1991** [comment]
@iandelible Answers to your questions below!

1. I think we need to have a discussion about where we want to put the obstruction check. I'm fine with placing it in the valid_move? methods if thats what the group would like to do. Originally I thought that check would happen in the move_to method we have within the piece class. Not sure what the pros and cons are to either implementation.
2. Yes, the middle statement is to check to see if moving diagonally is a valid option for the pawn. Since the pawn can only move diagonally during a capture move, I just checked to see if there was an opponent piece at the location.
3. Probably a very good idea haha.  I'll add in the test when I work on the changes from Jacob's comment.

**Comment by dmk1991** [comment]
Fixes the diagonal bug, adds is_obstructed check to remain in conformity with the other valid_move? methods, and fixes tests to confirm obstructed is working.

**Review by jacobwgillespie — APPROVED** [review]
LGTM!

**Pull request merged**